common/djangoapps/third_party_auth/lti.py

```diff
@@ -0,0 +1,195 @@
+
+    def get_user_details(self, response):
+        """
+        Retrieves user details from LTI parameters
+        """
+        details = {}
+        lti_params = response[LTI_PARAMS_KEY]
+
+        def add_if_exists(lti_key, details_key):
+            """
+            Adds LTI parameter to user details dict if it exists
+            """
+            if lti_key in lti_params and lti_params[lti_key]:
+                details[details_key] = lti_params[lti_key]
+
+        add_if_exists('email', 'email')
+        add_if_exists('lis_person_name_full', 'fullname')
+        add_if_exists('lis_person_name_given', 'first_name')
+        add_if_exists('lis_person_name_family', 'last_name')
+        return details
+
+    @classmethod
+    def get_validated_lti_params(cls, strategy):
+        """
+        Validates LTI signature and returns LTI parameters
+        """
+        request = Request(
+            uri=strategy.request.build_absolute_uri(), http_method=strategy.request.method, body=strategy.request.body
+        )
+        lti_consumer_key = request.oauth_consumer_key
+        (lti_consumer_valid, lti_consumer_secret, lti_max_timestamp_age) = cls.load_lti_consumer(lti_consumer_key)
+        current_time = calendar.timegm(time.gmtime())
+        return cls._get_validated_lti_params_from_values(
+            request=request, current_time=current_time,
+            lti_consumer_valid=lti_consumer_valid,
+            lti_consumer_secret=lti_consumer_secret,
+            lti_max_timestamp_age=lti_max_timestamp_age
+        )
+
+    @classmethod
+    def _get_validated_lti_params_from_values(cls, request, current_time,
+                                              lti_consumer_valid, lti_consumer_secret, lti_max_timestamp_age):
+        """
+        Validates LTI signature and returns LTI parameters
+        """
+
+        # Taking a cue from oauthlib, to avoid leaking information through a timing attack,
+        # we proceed through the entire validation before rejecting any request for any reason.
+        # However, as noted there, the value of doing this is dubious.
+
+        base_uri = normalize_base_string_uri(request.uri)
+        parameters = collect_parameters(uri_query=request.uri_query, body=request.body)
+        parameters_string = normalize_parameters(parameters)
+        base_string = construct_base_string(request.http_method, base_uri, parameters_string)
+
+        computed_signature = sign_hmac_sha1(base_string, unicode(lti_consumer_secret), '')
+        submitted_signature = request.oauth_signature
+
+        data = {parameter_value_pair[0]: parameter_value_pair[1] for parameter_value_pair in parameters}
+
+        def safe_int(value):
+            """
+            Interprets parameter as an int or returns 0 if not possible
+            """
+            try:
+                return int(value)
+            except ValueError:
+                return 0
+            except TypeError:
+                return 0
```

Comment:
Nit: you can collapse these into `except (ValueError, TypeError):` so that you don't have to repeat the `return`.